weave-js/src/components/Timestamp.tsx

```diff
@@ -38,6 +38,18 @@ const formatSmallTime = (then: moment.Moment): string | null => {
   const hours = Math.floor(diffMs / (1000 * 60 * 60));
   const days = Math.floor(diffMs / (1000 * 60 * 60 * 24));
 
+  const years = now.diff(then, 'years');
+  if (years > 0) {
+    const remainingDays = days - years * 365;
+    if (remainingDays < 7) {
```

Comment:
I don't think I'm following the logic here - if remainingDays was >= 7 wouldn't we still want to show a year diff?

Reply:
we would show 53 weeks instead.

Reply:
idk, I sort of expect Martin to take a pass at this when he gets back

Reply:
I guess we dont need that level of fidelity, but what about 13 months vs 12 months?

Reply:
I think some more comments or unit tests would be good - this function doesn't match my intuition, which is that the older something is the less I care about a precise number. What we get now is:
formatSmallTime(moment("2025-01-01")) => 3mo
formatSmallTime(moment("2024-01-01")) => 459d
formatSmallTime(moment("2021-01-01")) => 222w
And that feels wrong.

Reply:
hahahah great breakdown, i'm convinced. fixing

Reply:
<img width="673" alt="Screenshot 2025-04-04 at 12 53 21 PM" src="https://github.com/user-attachments/assets/e5d23474-271b-47c4-82ec-8c2e836ee6d6" />
<img width="670" alt="Screenshot 2025-04-04 at 12 53 32 PM" src="https://github.com/user-attachments/assets/d9b5f401-2d41-44ff-bfa4-b5e7b7503a8c" />
<img width="668" alt="Screenshot 2025-04-04 at 12 53 43 PM" src="https://github.com/user-attachments/assets/a2d2b9ad-d747-4b1f-a262-2d33290a1131" />

Okay, its simpler, and more intuitive now